### PR TITLE
[EHN] Including LMDA

### PR DESCRIPTION
**Pull request by bruAristimunha**
A model that mixes ShallowNet and EEGNet, it was published in NeuroImage. I started this adaptation in February, but I had forgotten to open the PR here.

### CODE REVIEW

**Comment by bruAristimunha** [comment]
Small benchmark results with cross-session:

image

**Comment by bruAristimunha** [comment]
Cross-Subject:

image

image

File: braindecode/models/lmda.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        torch.Tensor
+            Output tensor after applying depth-wise attention.
+
+        """
+        x_pool = self.adaptive_pool(x)
+        x_transpose = x_pool.transpose(-2, -3)
+        y = self.conv(x_transpose)
+        y = self.softmax(y)
+        y = y.transpose(-2, -3)
+        return y * self.num_channels * x
```
Comment (martinwimpff) [inline braindecode/models/lmda.py:67]:
```suggestion
        x_pool = self.adaptive_pool(x)  # D' x 1 x num_times
        x_transpose = x_pool.transpose(-2, -3)  # 1 x D' x num_times
        y = self.conv(x_transpose)  # 1 x D' x num_times
        y = self.softmax(y)  # 1 x D' x num_times
        y = y.transpose(-2, -3)  # D' x 1 x num_times
        return y * self.num_channels * x  # D' x num_channels x num_times
```

File: braindecode/models/lmda.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+            Input tensor of shape `(batch_size, n_chans, time)`.
+
+        Returns
+        -------
+        torch.Tensor
+            Output logits of shape `(batch_size, n_outputs)`.
+
+        """
+        x = self.ensure_dim(x)
+        x = torch.einsum("bdcw, hdc->bhcw", x, self.channel_weight)  #
```
Comment (martinwimpff) [inline braindecode/models/lmda.py:270]:
```suggestion
        x = self.channel_weight(x)
```
With the convolution layer above, we would simplify to this expression.

**Review by martinwimpff — COMMENTED** [review]
I haven't read the paper but just from my experience, three aspects are weird:
- spatial layer (channel_weight) before temporal conv 
- two spatial layers: channel_weight and spatial_conv
- depthwise separable convolutions are in the "wrong" order (starting with pw-conv). However, the ordering might still make sense if viewed differently, i.e., the pw-conv is effectively part of the block before. Still, the pw-conv of the spatial_conv would be unnecessary in this case.

**Pull request closed without merging**